src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqTransportBindingElement.cs

```diff
@@ -3,35 +3,38 @@
 
 using System;
 using System.Net;
+using CoreWCF.Channels.Configuration;
 using CoreWCF.Configuration;
 using CoreWCF.Queue.Common;
 using CoreWCF.Queue.Common.Configuration;
 using RabbitMQ.Client;
 
 namespace CoreWCF.Channels
 {
-    public class RabbitMqTransportBindingElement : QueueBaseTransportBindingElement
+    public sealed class RabbitMqTransportBindingElement : QueueBaseTransportBindingElement
     {
-        /// <summary>
-        /// Creates a new instance of the RabbitMQTransportBindingElement Class using the default protocol.
-        /// </summary>
+        private const long DefaultMaxMessageSize = 512 * 2 ^ 20;  // 512 MB, max message size as of RabbitMQ v3.8
+
```

Comment:
WCF normally uses a MaxReceivedMessageSize of 64K. This is to limit memory consumption by malicious clients. 512MB is a massive amount of memory to allocate and if your messages aren't expected to be that big, this could be a problem. It's typical for users to increase the max message size if they need to transfer large amounts of data. 64K is generally enough for simple RPC calls and usually only needs to be increased if data blobs are being sent.